tfutils/base.py

```diff
@@ -484,12 +497,18 @@ def save(self, train_res=None, valid_res=None, step=None, validation_only=False)
         rec['step'] = step
 
         if len(train_res) > 0:
-            # TODO: also include error rate of the train set to monitor overfitting
+            # TODO: also include error rate of the train set to monitor
+            # overfitting
             message = 'Step {} ({:.0f} ms) -- '.format(step, 1000 * duration)
-            msg2 = ['{}: {:.4f}'.format(k, v) for k, v in train_res.items() if k != 'optimizer' and k not in self.save_to_gfs]
+            msg2 = ['{}: {:.4f}'.format(k, v) for k, v in train_res.items()
+                    if k not in ['optimizer', 'grads', 'zero_grad'] and k not in self.save_to_gfs]
```

Comment:
is there some less "ad hoc" way of specifying this so that (e.g.) other code like this save method doesn't have to know about the details of how the optimizer is up?